src/util/util_watchdog.c

```diff
@@ -160,6 +160,9 @@ static void watchdog_fd_read_handler(struct tevent_context *ev,
               "[%d]: %s\n", ret, sss_strerror(ret));
         orderly_shutdown(1);
     }
+    if (getpid() == getpgrp()) {
+        kill(-getpgrp(), SIGUSR2);
+    }
```

Comment:
This watchdog code is in all processes and not only in backend(provider/sssd_be).
And PR says it should be done only in providers(sssd_be); because other processes cannot directly change online/offline status. i think that only reasonable way how to detect that we are in provider is to check global variable `debug_prg_name` (whis is also used in debug messages)

and I would also prefer sending signal `USR2` just to process itself. `kill(getpid(0, SIGUSR2)`

Reply:
@lslebodn, nice catch. Thanks for stepping in!